org.fcitx.Fcitx5.Addon.McBopomofo.yaml

```diff
@@ -50,8 +50,11 @@ modules:
     sources:
       - type: git
         url: https://github.com/openvanilla/fcitx5-mcbopomofo
-        tag: 2.7
-        commit: 6e3ab946f27368a10ee59ccde34e1eefdd65beed
+        tag: 2.8
+        commit: 1fbc7e1bbfee3ca3bec9549844a9674872f913c4
```

Comment:
I realized that I made a tagged release without properly updating the metadata… I'll make a 2.8.1 release with properly updated metadata soon. My apologies.

Reply:
@xlionjuan We just tagged the 2.8.1: https://github.com/openvanilla/fcitx5-mcbopomofo/releases/tag/2.8.1 ; could you please take a look when you have a chance? Thanks again!

Reply:
Consider just merge and test if the Flathub bot works 😄

![Screenshot_20241107-180052.png](https://github.com/user-attachments/assets/052967c8-0a6b-4d10-9adc-a9e05f80e701)

Reply:
The bot definitely worked, but I can't merge this to make the 2.8 release, since the fcitx5-mcbopomofo source at that tag does not have the proper metadata update (this is why you still saw "version 2.7" with `flakpak list`). This PR would need to be updated to use tag `2.8.1` and its corresponding commit to have a well-formed release.

Reply:
Isn't the bot just check to git tag just like my second commit?

The `flatpak` itself **isn't** check the update by version number, it is checking ostree, so the version number won't affect the distribution of the updates, it just shows wrong version number.

Reply:
Are interested in test the bot? If yes, merge it and wait few hours or tomorrow, if not, I will update the yaml.

Reply:
My understanding (especially after reading [its purpose here](https://github.com/flathub-infra/flatpak-external-data-checker?tab=readme-ov-file#changes-to-flatpak-manifests)) you're interested in validating whether the external data checker will send us new PRs when it discovers a new version of fcitx5-mcbopomofo, right?

(As a side note: do you really think the data checker will help us update fcitx5-mcbopomofo's metainfo.xml? The project doesn't use `metainfo.xml` per se, but uses [a template file](https://github.com/openvanilla/fcitx5-mcbopomofo/blob/master/org.fcitx.Fcitx5.Addon.McBopomofo.metainfo.xml.in) and produces `metainfo.xml` during the build process. I just wanted to make sure that we the fcitx5-mcbopomofo's maintainers all understand what to expect here.)

If that's indeed the case, then I'm also interested in seeing if it works, and I really appreciate your contribution. It's just that fcitx5-mcbopomofo 2.8 is a bad release—the version number in its fcitx5 module metadata isn't properly updated to 2.8. So was its flatpak metadata (`org.fcitx.Fcitx5.Addon.McBopomofo.metainfo.xml.in`).

So while I'm all for testing if `x-checker-data` will work, I just don't think we should test it with a bad release .

Yes, if I remember it correctly what I did with the previous flakpak versions, there should still be a manual release button I need to click on Flathub's build system to really publicly release 2.8, but I'd rather we use version 2.8.1—a well-formed version in the source repo—as the basis of the PR, and see if the next tagged version works. Does that make sense to you?

Reply:
I forgot Flathub suggested to upstream metainfo file, and you've did that, in this cace, you still need to make sure the version number on your original repo is updated correctly.


What Flathub bot do at here is help you update the git tag and git SHA automatically than push the PR, and just like other PRs, Flathub bot will build the release and post a command if test is needed, if everything works, just simply merge it.

Less work, less lazy, less delay.

> there should still be a manual release button I need to click on Flathub's build system

I haven't maintained Flathub apps before, so I don't know that, now I knows.

Reply:
> What Flathub bot do at here is help you update the git tag and git SHA automatically than push the PR, and just like other PRs, Flathub bot will build the release and post a command if test is needed, if everything works, just simply merge it.
>
> Less work, less lazy, less delay.

Indeed. I look forward to a near future where none of us needs to manually update the YAML again. I learned a lot from this PR. Thanks again for the contribution!

Reply:
Also a note here: while there is indeed a Release button in Flathub builder, it's designed for the maintainer to expedite the release process. Without any intervention, a successfully bulit stable version will be automatically published within a few hours. So merging anything in this repo does mean that it will publish a stable release of fcitx5-mcbpomofo flatpak.

Reply:
Oh so that button= If the build successful, don't ask me again, just release it.

Reply:
A Flatpak stable package is first built by a builder, and then the package is scheduled to be published by a separate publish job. I believe the Publish button on the first job gives maintainers like me the ability to expedite the publish process (by scheduling the publish job earlier)—or intervene it (for example, if we need to cancel the release at the last minute; in that case nothing will be published IIUC).

I thought I would need to manually click that button, but in the end it is really just a way to expedite the process. Like I said above, f a maintaner does nothing after a stable package is built, then a publish job is scheduled automatically, and the package will then be published within the next few hours.

Reply:
You can enable auto merge you want to(of course it will only merge if build successful.